synapse/storage/data_stores/main/state.py

```diff
@@ -363,24 +364,46 @@ def _background_remove_left_rooms_txn(txn):
             if not room_ids:
                 return True, set()
 
+            # exclude rooms where we have active members
             sql = """
                 SELECT room_id
-                FROM current_state_events
+                FROM local_current_membership
                 WHERE
                     room_id > ? AND room_id <= ?
-                    AND type = 'm.room.member'
                     AND membership = 'join'
-                    AND state_key LIKE ?
                 GROUP BY room_id
             """
 
-            txn.execute(sql, (last_room_id, room_ids[-1], "%:" + self.server_name))
-
+            txn.execute(sql, (last_room_id, room_ids[-1]))
```

Comment:
these changes are unrelated, but I thought I may as well simplify it while I was here.